pkg/reconciler/v1alpha1/revision/cruds.go

```diff
@@ -135,10 +135,12 @@ func (c *Reconciler) createService(ctx context.Context, rev *v1alpha1.Revision,
 func (c *Reconciler) checkAndUpdateService(ctx context.Context, rev *v1alpha1.Revision, sf serviceFactory, service *corev1.Service) (*corev1.Service, Changed, error) {
 	logger := logging.FromContext(ctx)
 
-	desiredService := sf(rev)
-
-	// Preserve the ClusterIP field in the Service's Spec, if it has been set.
-	desiredService.Spec.ClusterIP = service.Spec.ClusterIP
+	// Note: only reconcile the spec we set.
+	originalService := sf(rev)
```

Comment:
I think these got swapped. The Service returned by `sf` is what we want, and the `service` passed into `checkAndUpdateService` is the current state of things.

Reply:
That's my bad naming. `originalService` I meant the original service we want.